bot/exts/backend/sync/_cog.py

```diff
@@ -20,33 +21,39 @@ class Sync(Cog):
 
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
+        self.guild: Guild = None
+
 
     async def cog_load(self) -> None:
         """Syncs the roles/users of the guild with the database."""
         await self.bot.wait_until_guild_available()
 
-        guild = self.bot.get_guild(constants.Guild.id)
-        if guild is None:
-            return
+        self.guild = self.bot.get_guild(constants.Guild.id)
+        if self.guild is None:
+            raise ValueError("Could not fetch guild from cache, not loading sync cog.")
 
         attempts = 0
         while True:
             attempts += 1
-            if guild.chunked:
+            if self.guild.chunked:
                 log.info("Guild was found to be chunked after %d attempt(s).", attempts)
                 break
 
             if attempts == MAX_ATTEMPTS:
                 log.info("Guild not chunked after %d attempts, calling chunk manually.", MAX_ATTEMPTS)
-                await guild.chunk()
+                await self.guild.chunk()
                 break
 
             log.info("Attempt %d/%d: Guild not yet chunked, checking again in 10s.", attempts, MAX_ATTEMPTS)
             await asyncio.sleep(10)
+        create_task(self.sync())
+
+    async def sync(self) -> None:
+        await asyncio.sleep(10)  # Give time to other cogs starting up
```

Comment:
Which other cogs do we depend on? Or is it just to prevent ratelimits?

Reply:
only so many tasks can be ran at a time, so this just allows other cogs to startup instead of the syncing cog taking up all processing power. No dependencies.

Reply:
hmm, it shouldn't make much of a difference given the sync cog would mostly be waiting on network tasks so other things can run, are we sure this actually helps?